addons/point_of_sale/static/src/js/printers.js

```diff
@@ -77,7 +86,7 @@ var PrinterMixin = {
             // IoT box can't find a printer.
             if (!sendPrintResult || sendPrintResult.result === false) {
                 this.receipt_queue.length = 0;
-                return this.printResultGenerator.IoTResultError(sendPrintResult.printerErrorCode);
+                return this.printResultGenerator.IoTResultError(sendPrintResult);
```

Comment:
why was this changed ? i do wonder if it will not alter other flows

Reply:
I've changed this part because I need the whole `sendPrintResult` object to check if the `sendPrintResult.idenrtifier === undefined` (printers.js, L30). You're right, it may break the IoTResultError() override in /home/odoo/src/odoo/addons/pos_epson_printer/static/src/js/printers.js. I'm fixing this in my next commit, thanks! :1st_place_medal:

Reply:
@Jihaysse thanks for the details, indeed you will need it and your attempt to fix it is nice, however, there is a catch here.
If we were on master you could have proposed that, but as we are in stable version we should NOT change function arguments. Think about it, if someone override this function with custom code, the argument will now suddenly change on his side. Even if it is pretty unlikely in this case, it can happens. We will likely discuss it with the PoS/IoT team later on. But I guess it would be better in this case to add your parameter as an extra argument:
```suggestion
                return this.printResultGenerator.IoTResultError(sendPrintResult.printerErrorCode, sendPrintResult);
```
(and adapt below to use the second argument)

We do agree that it is a bit stupid as the second argument contains the first one. But this is the best tradeoff I can think of if we want to ensure stability